public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -74,6 +77,10 @@ export class PanelChrome extends Component<Props, State> {
         eventBus,
         onSeriesColorChange: this.onSeriesColorChange,
         onToggleSeriesVisibility: this.onSeriesVisibilityChange,
+        createAnnotation: this.onAnnotationCreate,
+        updateAnnotation: this.onAnnotationUpdate,
+        deleteAnnotation: this.onAnnotationDelete,
+        canAddAnnotations: () => Boolean(props.dashboard.meta.canEdit || props.dashboard.meta.canMakeEditable),
```

Comment:
These seem to follow a different naming pattern compared to the other functions (onSeriesColorChange etc)

Reply:
Wonder if we should make these optional, so that from a PanelRenderer (plugins that use TimeSeriesPanel) does not have to implement this. Looks like PanelRenderer does not wrap component in a PanelContextProvider. I guess we have to think about where that should live for plugins that want to re-use panels . 

The problem we have now is that ContextMenu and the Add Annotation menu item shows up in the Alert rule preview UI Graph (it is using PanelRenderer)

Reply:
These are optional now :)

Reply:
> The problem we have now is that ContextMenu and the Add Annotation menu item shows up in the Alert rule preview UI Graph (it is using PanelRenderer)

This is now handled in the panel. If ctx contains `canAddAnnotation` check and it returns false, the annotation creation is removed from context menu.